db/migrate/20181004181337_refactor_operations_to_have_mission_and_new_attributes.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class RefactorOperationsToHaveMissionAndNewAttributes < ActiveRecord::Migration[5.1]
+  def up
+    execute("DELETE FROM operations")
+    remove_column(:operations, :creator_old_id)
+    remove_column(:operations, :old_id)
+    rename_column(:operations, :description, :details)
+    add_column(:operations, :mission_id, :uuid)
```

Comment:
Whenever you add a foreign key you should add a foreign key constraint and an index. This checklist might be helpful: https://redmine.sassafras.coop/projects/sassafras-internal/wiki/Rails_Best_Practices

Search the project for add_foreign_key and add_index for the appropriate syntax.

Don't need to worry about them in the `down` migration, they will disappear when you remove the column.

Reply:
@smoyth A lot of the foreign keys in the db have `on_update: restrict`, and `on_delete: restrict`. Is this what we want for this foreign key as well? Is this assuming that the `has_many :operations, inverse_of: :mission, dependent: :destroy` will delete an operation when its mission is deleted?

Reply:
answered :)